src/3d/qgsabstractmaterialsettings.h

```diff
@@ -115,6 +115,11 @@ class _3D_EXPORT QgsAbstractMaterialSettings SIP_ABSTRACT
      */
     virtual QgsLineMaterial *toLineMaterial( const QgsMaterialContext &context ) const = 0 SIP_FACTORY;
 
+    /**
+     * Returns the parameters to be exported to .mtl file
+     */
+    virtual QMap<QString, QString> toExportParameters() const = 0 SIP_FACTORY;
```

Comment:
```suggestion
    virtual QMap<QString, QString> toExportParameters() const = 0;
```

The factory annotation is only required for methods which return a pointer along with ownership (or a list of pointers). This function is returning values directly, so it's not required here.